cluster/manifests/prometheus/statefulset.yaml

```diff
@@ -1,42 +1,60 @@
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: apps/v1beta1
+kind: StatefulSet
 metadata:
-  name: prometheus
-  namespace: kube-system
+  annotations:
   labels:
     application: prometheus
+  name: prometheus
+  namespace: kube-system
 spec:
-  replicas: 1
+  replicas: 2
+  selector:
+    matchLabels:
+      application: prometheus
+  serviceName: prometheus
   template:
     metadata:
       labels:
         application: prometheus
     spec:
       containers:
       - name: prometheus
-        image: prom/prometheus:latest
+        image: registry.opensource.zalan.do/teapot/prometheus:v2.1.0
         args:
         - "--config.file=/etc/prometheus/prometheus.yml"
         - "--storage.tsdb.path=/prometheus/"
         ports:
         - name: ingress-port
           containerPort: 8080
+          protocol: TCP
         resources:
           limits:
             cpu: 200m
-            memory: 200Mi
+            memory: 2000Mi
           requests:
             cpu: 25m
-            memory: 25Mi
+            memory: 200Mi
         volumeMounts:
         - name: prometheus-config-volume
-          mountPath: /etc/prometheus/
+          mountPath: /etc/prometheus
         - name: prometheus-storage-volume
-          mountPath: /prometheus/
+          mountPath: /prometheus
+        securityContext:
+          allowPrivilegeEscalation: false
       volumes:
         - name: prometheus-config-volume
           configMap:
             defaultMode: 420
             name: prometheus-conf
-        - name: prometheus-storage-volume
-          emptyDir: {}
+      securityContext:
+        runAsUser: 65534
+        fsGroup: 65534
```

Comment:
A comment on why this is needed would be could for the future :)

I guess `65534` -> `nobody`?

Reply:
Yes, but I will not change it, because if the nobody user get 42 as uid then this container can not read and write data anymore, because of uid miss match (media has 65534 and uid 42), which should be in theory changed, but better to be sure.